react-common/components/controls/TeachingBubble.tsx

```diff
@@ -0,0 +1,291 @@
+        updatePosition(bubble, top + "px", left + "px");
+        updatePosition(bubbleArrow, arrowTop + "px", arrowLeft + "px");
+    }
+
+    const positionAbove = () => {
+        top = targetBounds.top - bubbleBounds.height - margin;
+        left = targetBounds.left - (bubbleBounds.width - targetBounds.width) / 2;
+        arrowTop = top + bubbleBounds.height;
+        arrowLeft = targetBounds.left + (targetBounds.width - margin) / 2;
+        bubbleArrow.style.borderLeft = transparentBorder;
+        bubbleArrow.style.borderRight = transparentBorder;
+        bubbleArrow.style.borderTop = opaqueBorder;
+    }
+
+    const positionBelow = () => {
+        top = targetBounds.bottom + margin;
+        left = targetBounds.left - (bubbleBounds.width - targetBounds.width) / 2;
+        arrowTop = top - margin;
+        arrowLeft = targetBounds.left + (targetBounds.width - margin) / 2;
+        bubbleArrow.style.borderLeft = transparentBorder;
+        bubbleArrow.style.borderRight = transparentBorder;
+        bubbleArrow.style.borderBottom = opaqueBorder;
+    }
+
+    const positionLeft = () => {
+        top = targetBounds.top - (bubbleBounds.height - targetBounds.height) / 2;
+        left = targetBounds.left - margin;
+        arrowTop = top + (bubbleBounds.height - margin) / 2;
+        arrowLeft = targetBounds.left - margin;
+        bubbleArrow.style.borderLeft = opaqueBorder;
+        bubbleArrow.style.borderTop = transparentBorder;
+        bubbleArrow.style.borderBottom = transparentBorder;
+    }
+
+    const positionRight = () => {
+        top = targetBounds.top - (bubbleBounds.height - targetBounds.height) / 2;
+        left = targetBounds.right + margin;
+        arrowTop = top + (bubbleBounds.height - margin) / 2;
+        arrowLeft = targetBounds.right;
+        bubbleArrow.style.borderRight = opaqueBorder;
+        bubbleArrow.style.borderTop = transparentBorder;
+        bubbleArrow.style.borderBottom = transparentBorder;
+    }
+
+    const positionCenter = () => {
+        top = (targetBounds.height - bubbleBounds.height) / 2 + targetBounds.top;
+        left = (targetBounds.width - bubbleBounds.width) / 2 + targetBounds.left;
+    }
+
+    const updatePosition = (element: HTMLElement, top: string, left: string) => {
+        element.style.top = top;
+        element.style.left = left;
+    }
+
+    const closeLabel = lf("Close");
+    const backLabel = lf("Back");
+    const nextLabel = lf("Next");
+    const confirmLabel = lf("Got it");
+    const finishLabel = lf("Finish");
+    const hasPrevious = stepNumber > 1;
+    const hasNext = stepNumber < totalSteps;
+    const hasSteps = totalSteps > 1;
+
+    const classes = classList(
+        "teaching-bubble-container",
+        className
+    );
+
+    return ReactDOM.createPortal(<FocusTrap className={classes} onEscape={onClose}>
+        <div className="teaching-bubble-cutout"></div>
+        <div className="teaching-bubble-arrow"></div>
```

Comment:
```suggestion
        <div className="teaching-bubble-cutout" />
        <div className="teaching-bubble-arrow" />
```

nit; defining as self closing can be a little helpful to indicate 'nothing is intended to be inside here' / that it intentionally doesn't have any child divs; fine either way though